### PR TITLE
Add regexp filter for artifact name

### PR DESCRIPTION
**Pull request by EnricoMi**
Currently, either one artifact (by name) or all artifacts can be downloaded.

This adds option `filter`, which is used as a regular expression to match the artifact names. Only those artifacts where the name matches the regular expression are downloaded.

### CODE REVIEW

**Comment by dawidd6** [comment]
- `name` and `filter` inputs conflict with each other, what happens is someone specifies both?
- you need to add the input to `action.yml` too
- some CI job to test this new input would be nice

File: action.yml [hunk 1]
```diff
@@ -47,6 +47,10 @@ inputs:
   name:
     description: Artifact name (download all artifacts if not specified)
     required: false
+  name_is_regexp:
+    description: Treat artifact name as a regular expression and download only artifacts with matching names
+    required: false
+    default: "false"
```
Comment (dawidd6) [inline action.yml:53]:
```suggestion
```

Comment (dawidd6) [reply action.yml:53]: It's `false` or undefined by default.

Comment (dawidd6) [reply action.yml:53]: Okay, fine, then just remove the quotes please.

Comment (EnricoMi) [reply action.yml]: Removed. Technically, the type of `default` is string, not a boolean.

**Comment by dawidd6** [comment]
Nice work, thanks!

**Comment by EnricoMi** [comment]
Works, thanks for merging so quickly.

**Pull request merged**